src/lib.rs

```diff
@@ -196,6 +222,29 @@ named!(memory_read<&[u8], (u64, u64)>,
 named!(read_register<&[u8], u64>,
        preceded!(tag!("p"), hex_value));
 
+/// Helper for parse_thread_id that parses a single thread-id element.
+named!(parse_thread_id_element<&[u8], Id>,
+       alt_complete!(tag!("0") => { |_| Id::Any }
+                     | tag!("-1") => { |_| Id::All }
+                     | hex_value => { |val: u64| Id::Id(val as u32) }));
+
+/// Parse a thread-id.
+named!(parse_thread_id<&[u8], ProcessId>,
+       alt_complete!(parse_thread_id_element => { |pid| ProcessId { pid: pid, tid: Id::Any } }
+                     | preceded!(tag!("p"),
```

Comment:
I was imagining some sort of implementation where the `Handler` indicated whether it supported multiprocess, and we adjusted the parsing of thread IDs accordingly.  But this seems more elegant, and it seems unlikely that GDB (or other debug agents...) will be sending multiprocess-aware IDs even if we're only debugging a single process?

Reply:
I think clients aren't supposed to unless both sides negotiate multi-process.

We could check this here or in the caller and return an error in that case.

Reply:
Looking through the GDB docs while filing issues, my understanding is that GDB will send simple numbers, unless we've negotiated support for multiprocess, in which case it will always send `pX.Y`.  Is that correct?

I think the current implementation is fine.  We can work out something more sophisticated later.

Reply:
> Is that correct?

I believe so.